iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -456,9 +458,19 @@ public boolean isValidMetadataTable(String metaTableName) {
   public URI getURIForAuth(org.apache.hadoop.hive.metastore.api.Table hmsTable) throws URISyntaxException {
     String dbName = hmsTable.getDbName();
     String tableName = hmsTable.getTableName();
-    return new URI(ICEBERG_URI_PREFIX + dbName + "/" + tableName);
+    Optional<String> locationProperty = SessionStateUtil.getProperty(conf, hive_metastoreConstants.META_TABLE_LOCATION);
+    if (locationProperty.isPresent()) {
+      // this property is set during the create operation before the hive table was created
+      // we are returning a dummy iceberg metadata file
+      return new URI(ICEBERG_URI_PREFIX + locationProperty.get() + "/metadata/dummy.metadata.json");
```

Comment:
Do we always set the location?
Would it be worthwhile to check this?

Reply:
yes, we are setting it. That is why I added that additional code change to the SemanticAnalyzer

Reply:
So a `Precondition` check could be ok here?

Reply:
sure, that can be done